stats.go

```diff
@@ -60,84 +60,156 @@ type counter struct {
 	prev        int64
 	curr        int64
 	cachedCount CachedCount
+	tracker     *metricExpiryTracker
 }
 
-func newCounter(cachedCount CachedCount) *counter {
-	return &counter{cachedCount: cachedCount}
+func newCounter(cachedCount CachedCount, name string, scope *scope) *counter {
+	return &counter{
+		cachedCount: cachedCount,
+		tracker: &metricExpiryTracker{
+			scope:          scope,
+			name:           name,
+			lastUpdateUnix: globalNow().Unix(),
+		},
+	}
 }
 
 func (c *counter) Inc(v int64) {
 	atomic.AddInt64(&c.curr, v)
+
+	if c.tracker.expiredAndUpdated() {
```

Comment:
can return early here (and for the other types),
```
atomic.AddInt64(&c.curr, 1)

if !c.tracker.expiredAndUpdated() {
  return
}

// expiry handling
```

Reply:
What happens when a manager codes :P